#ifndef EDITRECORDDIALOG_H
#define EDITRECORDDIALOG_H
#include <QDialog>
#include <QFileDialog>
#include "record.h"

namespace Ui {
    class EditRecordDialog;
};


class EditRecordDialog : public QDialog {
    Q_OBJECT
    Record* record;
    bool sigSet;
    QByteArray selectedData;
    bool checkFields();
public:
    explicit EditRecordDialog(QWidget* parent = Q_NULLPTR);
    explicit EditRecordDialog(Record* record, QWidget* parent = Q_NULLPTR);
    ~EditRecordDialog();
    Record* getRecord();
public slots:
    void checkAccept();
    void chooseSig();
    void changedPrefixLen(int value);
private:
    Ui::EditRecordDialog* ui;
	
};

#endif